app/src/main/java/de/tum/in/tumcampusapp/component/other/generic/drawer/DrawerMenuHelper.kt

```diff
@@ -1,116 +1,151 @@
+        get() = navigationView.menu
 
-        val allItems = mutableListOf<SideNavigationItem>()
-        val navigationMenu = navigationView.menu
+    private val allItems = mutableListOf<NavItem>()
 
-        navigationMenu.add(activity, HOME, drawerBundle)
-        allItems.add(HOME)
+    fun populateMenu() {
+        val hasTumOnlineAccess = AccessTokenManager.hasValidAccessToken(activity)
+        val isChatEnabled = Utils.getSettingBool(activity, Const.GROUP_CHAT_ENABLED, false)
+        val isEmployeeMode = Utils.getSettingBool(activity, Const.EMPLOYEE_MODE, false)
+
+        navigationMenu.clear()
+        allItems.clear()
+
+        navigationMenu += HOME
+        allItems += HOME
 
         val myTumMenu = navigationMenu.addSubMenu(R.string.my_tum)
-        if (hasTUMOAccess) {
-            var items = MY_TUM.filterNot {
-                it.needsChatAccess && !chatEnabled }
-            if (employeeMode) {
-                items = items.filterNot { it.hideForEmployees }
-            }
-            items.forEach {
-                myTumMenu.add(activity, it, drawerBundle)
-                allItems.add(it)
+        if (hasTumOnlineAccess) {
+            val candidates = MY_TUM
+                .filterNot { !isChatEnabled && it.needsChatAccess }
+                .filterNot { isEmployeeMode && it.hideForEmployees }
+
+            for (candidate in candidates) {
+                myTumMenu += candidate
+                allItems += candidate
             }
         }
 
-        // General information which mostly can be used without a TUMonline token
-        val commonTumMenu = navigationMenu.addSubMenu(R.string.common_info)
-        COMMON_TUM
-                .filterNot { it.needsTUMOAccess && !hasTUMOAccess }
-                .forEach {
-                    commonTumMenu.add(activity, it, drawerBundle)
-                    allItems.add(it)
-                }
+        val generalMenu = navigationMenu.addSubMenu(R.string.common_info)
+        val generalCandidates = GENERAL.filterNot { it.needsTUMOAccess && !hasTumOnlineAccess }
+        for (candidate in generalCandidates) {
+            generalMenu += candidate
+            allItems += candidate
+        }
 
-        // App related menu entries
         val aboutMenu = navigationMenu.addSubMenu(R.string.about)
-        ABOUT.forEach {
-            aboutMenu.add(activity, it)
-            allItems.add(it)
+        for (item in ABOUT) {
+            aboutMenu += item
+            allItems += item
         }
 
-        // Highlight the currently selected activity.
-        val currentIndex = allItems
-                .indexOfFirst { it.activity == activity::class.java }
-                .takeIf { it != -1 }
-        currentIndex?.let { index ->
-            navigationMenu.allItems[index].apply {
-                isCheckable = true
-                isChecked = true
+        highlightCurrentItem()
+    }
+
+    fun findNavItem(menuItem: MenuItem): NavItem {
```

Comment:
Can we do this better than linearely searching through all the items?
Maybe do a Map or something?